BeerProject/src/test/java/wawer/kamil/beerproject/controllers/UserControllerTest.java

```diff
@@ -1,125 +1,248 @@
-        assertEquals(HttpStatus.CREATED,controller.createNewUser(userDTO).getStatusCode());
+    @DisplayName("Test - should return response entity with updated user")
+    public void should_return_response_entity_with_updated_user() throws NoContentException {
+        //given
+        when(userService.updateUser(USER_ID, userRequest)).thenAnswer(invocation -> {
+            UserRequest argument = invocation.getArgument(1, UserRequest.class);
+            UserResponse userResponse = new UserResponse();
+            userResponse.setId(USER_ID);
+            userResponse.setUsername(argument.getUsername());
+            userResponse.setPassword(argument.getPassword());
+            userResponse.setEmail(argument.getEmail());
+            userResponse.setGrantedAuthorities(argument.getGrantedAuthorities());
+            userResponse.setEnabled(true);
+            userResponse.setCredentialsNonExpired(true);
+            userResponse.setAccountNonLocked(true);
+            userResponse.setAccountNonExpired(true);
+            return userResponse;
+        });
+
+        //when
+        ResponseEntity<UserResponse> newUserResponseEntity = controller.updateUser(USER_ID, userRequest);
+
+        //than
+        assertThat(newUserResponseEntity).isNotNull();
+        assertThat(newUserResponseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(newUserResponseEntity.getBody()).isNotNull();
+        assertThat(newUserResponseEntity.getBody().getId()).isEqualTo(1L);
+        assertThat(newUserResponseEntity.getBody().getUsername()).isEqualTo("user");
+        assertThat(newUserResponseEntity.getBody().getPassword()).isEqualTo("user");
+        assertThat(newUserResponseEntity.getBody().getEmail()).isEqualTo("test@email.com");
+        assertThat(newUserResponseEntity.getBody().isAccountNonExpired()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isAccountNonLocked()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isCredentialsNonExpired()).isTrue();
+        assertThat(newUserResponseEntity.getBody().isEnabled()).isTrue();
     }
 
     @Test
-    public void should_return_saved_user_which_equals_to_user_updated_by_controller() throws NoContentException {
-        when(mapper.map(userDTO, User.class)).thenReturn(user);
-        when(service.updateUser(ID, user)).thenReturn(user);
-        when(mapper.map(user, UserDTO.class)).thenReturn(userDTO);
-        when(user.getId()).thenReturn(ID);
-        assertEquals(ResponseEntity.status(HttpStatus.OK).body(userDTO),controller.updateUser(ID, userDTO));
+    @DisplayName("Test - should throw exception when user not found during update")
+    public void should_throw_exception_when_user_not_found_during_update() throws NoContentException {
+        //given
+        when(userService.updateUser(USER_ID, userRequest)).thenThrow(NoContentException.class);
+
+        //when
+        assertThatThrownBy(() -> controller.updateUser(USER_ID, userRequest));
     }
```

Comment:
może tam gdzie assert to zrobić //then bo assert to już koniec testu, i zamiast //given to //when

Reply:
jw

Reply:
jw